sql-plugin/src/main/scala/com/nvidia/spark/rapids/RapidsMeta.scala

```diff
@@ -807,6 +809,16 @@ abstract class BaseExprMeta[INPUT <: Expression](
   }
 
   final override def tagSelfForGpu(): Unit = {
+    val isConstantFoldingOff =
+      SQLConf.get.optimizerExcludedRules
+        .contains("org.apache.spark.sql.catalyst.optimizer.ConstantFolding")
+    if (isConstantFoldingOff) {
+      val exp = wrapped.asInstanceOf[Expression]
```

Comment:
is this cast needed?

Reply:
no, I removed and  tested against Spark 3.0.0